### DISCUSSION TITLE
sbt-svelte: the best way to use Svelte with Playframework

### DISCUSSION THREAD

**Discussion by tanin47** [playframework/playframework#11922] [General]
Hello!

I've made sbt-svelte which I believe is the best way to use Svelte with Playframework because it doesn't require you to run a separate npm process; you'd also store your Svelte files in the natural place, which is `/app/assets`.

sbt-svelte hooks into Playframework's asset generation mechanism. This means you can just run `sbt run`, and it just works. All your changes will be automatically detected by Play and compiled.

This requires Webpack though since Play doesn't know how to compile Svelte.

As a side note, I also built [sbt-vuefy](https://github.com/GIVESocialMovement/sbt-vuefy) (for Vue), which utilizes the same mechanism, a few years back, and it has been working well.

I'd love for y'all to try it out.

You can read more about it here: https://tanin.nanakorn.com/main/2023/07/02/use-svelte-with-playframework

You can see the repo here: https://github.com/tanin47/sbt-svelte -- there is a test project inside as well.

Thank you!

**Comment by mkurz** [comment]
Awesome, thanks for sharing! Would be good to add the project(s) to https://github.com/playframework/playframework/blob/main/documentation/manual/ModuleDirectory.md - can you open a pull request for that?
Great work!

**Reply by tanin47** [reply]
Thank you for the recommendation. I'll do that.

What is the criteria of adding a project there?

I have another project that I would love to add it there: https://github.com/GIVESocialMovement/play-json-form (an improvement to Play's form). I was hoping to improve Play's form someday but haven't got time haha.

**Reply by mkurz** [reply]
> What is the criteria of adding a project there?

No recommendation, just put it in a section where it fits the most. If the other project(s) are usuable with current stable 2.8.x you can add them as well of course.
